src/libPMacc/include/traits/GetNComponents.hpp

```diff
@@ -35,7 +35,11 @@ namespace traits
  * \return \p ::value as public with number of components (uint32_t)
  */
 template<typename T_Type, bool T_IsFundamental = boost::is_fundamental<T_Type>::value>
-struct GetNComponents;
+struct GetNComponents
+{
+    PMACC_CASSERT_MSG_TYPE( __GetNComponents_is_not_defined_for_this_type, T_Type, false ); 
+    static constexpr uint32_t dim = 0;
```

Comment:
@psychocoderHPC Why do you need this integer variable here and why was it not needed before?